### PR TITLE
DnD file loader

### PR DESCRIPTION
**Pull request by ivn-cote**

### CODE REVIEW

File: client/elements/book-form.jsx [hunk 1]
```diff
@@ -17,6 +18,7 @@ export default React.createClass({
           <TextField name="title" floatingLabelText="Book title"/><br/>
           <TextField name="author" floatingLabelText="Book author"/><br/>
           <p><label htmlFor="form_file">File</label><input id="form_file" type="file" name="file"/></p>
+          <XHRUploader url='http://localhost:3000/api/uploadfile' automaxFiles={1} />
```
Comment (ignovak) [inline client/elements/book-form.jsx]: Почему не `url='/api/uploadfile'`?

**Comment by ignovak** [comment]
ok

File: client/elements/book-form.jsx [hunk 2]
```diff
@@ -10,24 +11,70 @@ import RaisedButton from 'material-ui/lib/raised-button';
...
     }
   },
 
+  // TODO: ajax-submit
   render() {
     return (
       <div>
         <h2>Upload a new book</h2>
-        <form className="bookForm">
+        <MyUploader url={accessToken} fieldName="book" auto />
```
Comment (ivn-cote) [inline client/elements/book-form.jsx]: `<MyUploader url={accessToken} fieldName="book" auto callback={this.onFileLoaded} />`

File: client/elements/book-form.jsx [hunk 3]
```diff
@@ -10,24 +11,70 @@ import RaisedButton from 'material-ui/lib/raised-button';
...
+}
+
+// TODO: setup correct access_token
+const accessToken = '/bapi/books?access_token=kGWNoGt8LNXlzYLAJnzdrFQ1rnnIERmGooE89JMM8LbQg6vJGuSdH2iUTBuRF5n3';
+
 export default React.createClass({
   getDefaultProps() {
     return {
     }
   },
```
Comment (ivn-cote) [inline client/elements/book-form.jsx:84]:
```
onFileLoaded() {
this.setState({ isSubmitAvailabale: true });
}
```

File: client/elements/book-form.jsx [hunk 4]
```diff
@@ -10,24 +11,70 @@ import RaisedButton from 'material-ui/lib/raised-button';
...
+      xhr.open('POST', this.props.url, true);
+      xhr.send(formData);
+      this.xhrs[file.index] = xhr;
+
+      // Our code
+      // Add callback on successful file upload
+      xhr.onreadystatechange = () => {
+        if (xhr.readyState == 4 && xhr.status === 200) {
+          // TODO: here we have to update form action and enable submit button
+          console.log(JSON.parse(xhr.response));
```
Comment (ivn-cote) [inline client/elements/book-form.jsx]: this.props.callback(xhr.response)

**Pull request merged**